yt/frontends/ramses/field_handlers.py

```diff
@@ -223,6 +224,14 @@ def any_exist(cls, ds):
 
     @classmethod
     def detect_fields(cls, ds):
+        # Important: here we also detect that the dataset is the same
+        # object.
+        # This is to force the redetection of the fields when changing
+        # dataset.
+        if cls.detected_fields and cls.ds == ds:
+            return cls.detected_fields.copy()
+        cls.ds = ds
```

Comment:
Why not make this cache more than one file? I could imagine a dict mapping from fullly qualified filenames to a detected fields list.

I'm also not really a fan of having this cache live as an attribute of the `FieldFileHandler` class itself as that seems to me a strange place to store a global variable. Better to have some singleton defined at module level.